controllers/session/session_controller_int_test.go

```diff
@@ -284,6 +284,103 @@ var _ = Describe("Complete session manipulation", func() {
 			})
 		})
 
+		Context("when calculating components readiness", func() {
+			It("should all show ready on success", func() {
+				req1 := reconcile.Request{
+					NamespacedName: types.NamespacedName{
+						Name:      "test-session1",
+						Namespace: "test",
+					},
+				}
+				// Given - create first ref
+				res1, err := controller.Reconcile(context.Background(), req1)
+				Expect(err).ToNot(HaveOccurred())
+				Expect(res1.Requeue).To(BeFalse())
+
+				// Then - all components should be success
+				session := get.Session("test", "test-session1")
+
+				Expect(*session.Status.State).To(Equal(v1alpha1.StateSuccess))
+				Expect(session.Status.Readiness.Components.Pending).To((HaveLen(0)))
+				Expect(session.Status.Readiness.Components.UnReady).To((HaveLen(0)))
+				Expect(session.Status.Readiness.Components.Ready).ToNot((HaveLen(0)))
+			})
+
+			It("should show pending before execution and missing modification, and unready on unsuccess", func() {
+				testLocator := func(ctx model.SessionContext, ref model.Ref, store model.LocatorStatusStore, report model.LocatorStatusReporter) error {
```

Comment:
Maybe we should move these `func`s outside of the test itself? Might improve readability in my opinion.

Reply:
As long as it's only used here and it contains the // comments which are related to the setup of the test.. then I don't see how jumping around for assertions and logic helps readability.

Reply:
Maybe it's just me and the way how I see the code. 1:1, nothing can be done here :)